moneyed/classes.py

```diff
@@ -26,49 +28,61 @@ class Currency:
     canonical name, and countries the currency is used in.
     """
 
-    def __init__(self, code="", numeric=None, sub_unit=1, name=None, countries=None):
-        self.code = code
-        self.numeric = numeric
-        self.sub_unit = sub_unit
-        if name is not None:
-            self.name = name
-        if countries is not None:
-            self.countries = countries
-
-    def __hash__(self):
+    def __init__(
+        self,
+        code: str = "",
+        numeric: Optional[str] = None,
+        sub_unit: int = 1,
+        name: Optional[str] = None,
+        countries: Optional[List[str]] = None,
+    ) -> None:
+        self.code: Final = code
+        self.numeric: Final = numeric
+        self.sub_unit: Final = sub_unit
+        self._name: Final = name
+        self._countries: Final = countries
+
+    def __hash__(self) -> int:
         return hash(self.code)
 
-    def __eq__(self, other):
-        return type(self) is type(other) and self.code == other.code
+    def __eq__(self, other: object) -> bool:
+        return (
+            type(self) is type(other)
+            and self.code == other.code  # type: ignore[attr-defined]
```

Comment:
Oh, cool! I didn't know that these comments could be for a specific rule

Reply:
Yeah it's very useful! A nice thing is that when configuring `warn_unused_ignores` mypy will error for comments like this that isn't used (e.g. there is no error at that line), so they sort of clean up themselves as mypy improves too.